.github/workflows/lint.yml

```diff
@@ -1,8 +1,6 @@
 name: lint
 
 on:
-  push:
-    branches: [main, "release/**"]
   pull_request:
     branches: [main, "release/**"]
   workflow_dispatch:
```

Comment:
I think it's worth adding a conditional on the `goheader` step to skip it on manual dispatch:

`if: ${{ github.event_name == 'pull_request' }}`

Reply:
actually, let's run lint as before and just have `if: ${{ github.event_name == 'pull_request' }}` for goheader - this will also help me merge jobs together (doing this in another branch now)

Reply:
SGTM